lib/facter/docker.rb

```diff
@@ -68,8 +68,8 @@ def interfaces
 Facter.add(:docker_version) do
   setcode do
     if Facter::Core::Execution.which('docker')
-      value = Facter::Core::Execution.exec(
-        "#{docker_command} version --format '{{json .}}'", timeout: 90
+      value = Facter::Core::Execution.execute(
+        "#{docker_command} version --format '{{json .}}'", options: {limit: 90}
```

Comment:
Can someone who knows ruby can confirm that this is how to do it? Whatever I pass as parameter to this function is accepted.

Also, while facter’s doc says the option is called `time_limit`, [it seems to actually be called `limit` in the code](https://github.com/puppetlabs/facter/blob/187a8687cb189c0e84a8f3089f89b0ccce42523a/lib/facter/custom_facts/core/execution/base.rb#L117).

Reply:
I guess the documentation is right and it's a bug in the code, I'll open a PR to address this :-)

There are a few more issues.  I opened #773 to address them.